Sources/CoffeeAddictsLibrary/Helpers/CoffeeShopsCSVParser.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  File.swift
+//  
+//
+//  Created by Mihai Garda Popescu on 23.07.2021.
+//
+
+import Foundation
+import CoreLocation
+
+typealias CoffeeShopsResult = ([CoffeeShop], ParsingStatus) -> Void
+
+class CoffeeShopsCSVParser {
+    static func parseCSVString(_ string: String, completion: @escaping CoffeeShopsResult) {
+        let parsedCSV: [[String]] = string.components(separatedBy: "\n").map{ $0.components(separatedBy: ",") }
+        
+        guard parsedCSV.count > 0 else {
+            completion([], .CSVFileInvalid)
+            return
+        }
+        
+        let coffeeShops = CoffeeShopsCSVParser.coffeeShopsFromParsedCSV(parsedCSV)
+        if coffeeShops.count == 0 {
+            completion([], .CSVFileEmptyResult)
+        }
+        
+        completion(coffeeShops, .CSVParsingSuccess)
+    }
+    
+    static func coffeeShopsFromParsedCSV(_ parsedCSV: [[String]]) -> [CoffeeShop] {
+        var coffeeShops: [CoffeeShop] = []
+        
+        for line in parsedCSV {
+            if line.count != 3 {
```

Comment:
I have a very particular linter that screams 'magic number' here, do you know about this code smell? And what we can do to avoid it?

Reply:
> I have a very particular linter that screams 'magic number' here, do you know about this code smell? And what we can do to avoid it?

Totally agree, I’d move it into a ‘let expectedNumberOfLineParameters = 3’ variable.

Note to self: Add a linter from the beginning :) 

Which linter are you using, by the way?

Reply:
Not sure what the options are in swift, I don't actively develop there :) but surely there must be one.

Reply:
Got it, thx, I know a few ones, I thought you used one in particular